cpp/oneapi/dal/table/homogen.hpp

```diff
@@ -119,18 +121,39 @@ class ONEDAL_EXPORT homogen_table : public table {
     }
 #endif
 
-public:
+    /// Creates a new ``homogen_table`` instance from an array.
+    /// The created table shares data ownership with the given array.
+    ///
+    /// @tparam Data        The type of elements in the data block that will be stored into the
+    ///                     table. The table initializes data types of metadata with this data type.
+    ///                     The feature types should be set to default values for :literal:`Data` type:
+    ///                     contiguous for floating-point, ordinal for integer types. The :literal:`Data`
+    ///                     type should be at least :expr:`float`, :expr:`double` or :expr:`std::int32_t`.
+    ///
+    /// @param data         The array that stores a homogeneous data block.
+    /// @param row_count    The number of rows in the table.
+    /// @param column_count The number of columns in the table.
+    /// @param layout       The layout of the data. Should be :literal:`data_layout::row_major` or
+    ///                     :literal:`data_layout::column_major`.
+    template <typename Data>
+    static homogen_table wrap(const array<Data>& data,
+                              std::int64_t row_count,
+                              std::int64_t column_count,
+                              data_layout layout = data_layout::row_major) {
+        return homogen_table{ data, row_count, column_count, layout };
+    }
+
     /// Creates a new ``homogen_table`` instance with zero number of rows and columns.
     /// The :expr:`kind` is set to``homogen_table::kind()``.
-    /// All the properties should be set to default values (see the Properties section).
```

Comment:
Maybe to rephrase, but not to remove completely?

Reply:
I'd also remove the phrase about kind. It's not clear how to "set default value" since we decided stop using properties.
However, it's important to understand behavior if "empty" table is created. I suggest to mention it in the method descriptor. I'll update `get_data` description.